tests/models/opt/test_modeling_opt.py

```diff
@@ -291,7 +292,7 @@ def test_generate_fp16(self):
         input_ids = input_dict["input_ids"]
         attention_mask = input_ids.ne(1).to(torch_device)
         model = OPTForCausalLM(config).eval().to(torch_device)
-        if torch_device == "cuda":
+        if is_torch_fp16_available_on_device(torch_device):
```

Comment:
This test should probably just be skipped if the device does not support fp16 (alternatively adding `@require_torch_fp16`).

Reply:
cc @arsalanu